plugins/fapolicyd.c

```diff
@@ -0,0 +1,212 @@
+#include "system.h"
+
+#include <rpm/rpmts.h>
+#include <rpm/rpmlog.h>
+#include "lib/rpmplugin.h"
+
+#include <fcntl.h>
+#include <errno.h>
+#include <unistd.h>
+#include <sys/stat.h>
+
+struct fapolicyd_data {
+    int fd;
+    int installed_package;
+    long changed_files;
+    const char * fifo_path;
+};
+
+static struct fapolicyd_data fapolicyd_state = {
+    .fd = -1,
+    .installed_package = -1,
+    .changed_files = 0,
+    .fifo_path = "/run/fapolicyd/fapolicyd.fifo",
+};
+
+static rpmRC open_fifo(struct fapolicyd_data* state)
+{
+    int fd = -1;
+    struct stat s;
+
+    fd = open(state->fifo_path, O_RDWR);
+    if(fd == -1) {
+        rpmlog(RPMLOG_DEBUG, "Open: %s -> %s\n", state->fifo_path, strerror(errno));
+        goto bad;
+    }
+
+    if (stat(state->fifo_path, &s) == -1) {
+        rpmlog(RPMLOG_DEBUG, "Stat: %s -> %s\n", state->fifo_path, strerror(errno));
+        goto bad;
+    } else {
+        if (!S_ISFIFO(s.st_mode)) {
+            rpmlog(RPMLOG_DEBUG, "File: %s exists but it is not a pipe!\n", state->fifo_path);
+            goto bad;
+        }
+
+        /* keep only file's permition bits */
+        mode_t mode = s.st_mode & ~S_IFMT;
+
+        /* we require pipe to have 0660 permission */
+        if (mode != 0660) {
+        
+            rpmlog(RPMLOG_ERR, "File: %s has %o instead of 0660 \n",
+                    state->fifo_path,
+                    mode );
+
+            goto bad;
```

Comment:
Using empty lines for grouping and as visual aid is totally okay, but these two seem redundant/to the contrary to me at least.